azurerm/internal/services/network/virtual_hub_resource.go

```diff
@@ -142,10 +144,23 @@ func resourceArmVirtualHubCreateUpdate(d *schema.ResourceData, meta interface{})
 		return fmt.Errorf("Error waiting for creation of Virtual Hub %q (Resource Group %q): %+v", name, resourceGroup, err)
 	}
 
-	resp, err := client.Get(ctx, resourceGroup, name)
+	// Hub returns provisioned while the routing state is still "provisining". This might cause issues with following hubvnet connection operations.
+	// https://github.com/Azure/azure-rest-api-specs/issues/10391
+	// As a workaround, we will poll the routing state and ensure it is "Provisioned".
+	timeout, _ := ctx.Deadline()
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"Provisioning"},
+		Target:       []string{"Provisioned", "Failed"},
+		Refresh:      virtualHubCreateRefreshFunc(ctx, client, resourceGroup, name),
+		PollInterval: 5 * time.Second,
```

Comment:
this'll cause (rate-limiting) issues for other resources - we likely want this set to `15` at a minimum (although likely 30) with a "Occurences" of 3 to confirm it's stable?